okapi-common/src/main/java/org/folio/okapi/common/ChattyResponsePredicate.java

```diff
@@ -0,0 +1,433 @@
+package org.folio.okapi.common;
+
+import static io.vertx.ext.web.client.predicate.ResponsePredicate.create;
+
+import io.vertx.core.impl.NoStackTraceThrowable;
+import io.vertx.ext.web.client.predicate.ErrorConverter;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import java.util.List;
+
+/**
+ * The same as {@link ResponsePredicate} but with response body included in the Exception message.
+ *
+ * <p>The body length is limited to 500 Unicode code units by removing characters in the middle.
+ */
+public final class ChattyResponsePredicate {
+  private static final int MAX_LENGTH = 500;
+
+  /**
+   * Concatenate the result message and the response body, and return an Exception with
+   * the concatenation result as message.
+   *
+   * @see ResponsePredicate#create(java.util.Function, ErrorConverter)
+   */
+  public static final ErrorConverter CONVERTER = ErrorConverter.createFullBody(result -> {
+    var body = result.response().bodyAsString();
+    if (body != null && body.length() > MAX_LENGTH) {
+      body = body.substring(0, MAX_LENGTH / 2) + "…"
+          + body.substring(body.length() - (MAX_LENGTH - 1) / 2);
+    }
```

Comment:
"Removing characters from the middle" (from the comment above)... Can you explain why you want to "remove characters from the middle". Is it because there is an assumption that the middle will contain extraneous information, but not the beginning or end?

Reply:
I see the purpose but I wonder if it should be the default behavior to truncate this middle part. It seems like a premature optimization perhaps and a bit arbitrary to base it off of character count. Could it be perhaps made a boolean option of the class such that users of `ChattyResponsePredicate` decide whether to reduce the body size if they know their error body is going to be large?

But this is just an idea for you to consider. I don't see this as a major problem .